### ISSUE TITLE
checkboxes in the table with selection example are missing labels

### ISSUE THREAD

**Issue by mfairchild365** [angular/material2#15021]
#### What is the expected behavior?
The purpose of the checkboxes in the 'table with selection' example must be conveyed to users of screen reader software.

#### What is the current behavior?
The purpose of the checkboxes in the 'table with selection' example is not conveyed to users of screen reader software.

#### What are the steps to reproduce?
1. Using either of the following screen readers
   - NVDA 2018.4.1 + Firefox 64
   - VoiceOver for IOS (12.1.3) + Safari
2. Open the [table example](https://material.angular.io/components/table/examples)
3. Navigate to one of the checkboxes in the buttons in the 'table with selection' example.
   - In NVDA you can press the tab key until you reach a checkbox
   - In VoiceOver for iOS you can swipe right until you reach a checkbox
4. Observe that the purpose of the checkbox is not announced by the screen reader.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
- Angular Material Example 7.2.2 located at https://material.angular.io/components/
- NVDA 2018.4.1 + FireFox 64
- VoiceOver for IOS (12.1.3) + Safari (12.1.3)

#### Is there anything else we should know?
Each checkbox must have a meaningful programmatic label.
